components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx

```diff
@@ -0,0 +1,690 @@
+    } = useGetNetworkQuery({
+      chainId: transaction.chainId,
+      coin: txCoinType
+    })
+
+    const networkAsset = React.useMemo(() => {
+      return makeNetworkAsset(txNetwork)
+    }, [txNetwork])
+
+    const {
+      data: combinedTokensList,
+      isLoading: isLoadingTokens
+    } = useGetCombinedTokensListQuery()
+
+    const {
+      data: accountInfosRegistry =
+      accountInfoEntityAdaptorInitialState,
+    } = useGetAccountInfosRegistryQuery(undefined)
+
+    const { account } = useAccountQuery(transaction.fromAccountId)
+
+    // memos & computed from queries
+    const txToken = findTransactionToken(
+      transaction,
+      combinedTokensList
+    )
+
+    const recipientLabel = getAddressLabel(
+      recipient,
+      accountInfosRegistry
+    )
+
+    const senderLabel = getAccountLabel(
+      transaction.fromAccountId,
+      accountInfosRegistry
+    )
+
+    const approvalTargetLabel = getAddressLabel(
+      approvalTarget,
+      accountInfosRegistry
+    )
+
+    const {
+      buyToken,
+      sellToken,
+      buyAmount,
+      sellAmount,
+      buyAmountWei
+    } = React.useMemo(() => {
+      return transaction.txType === BraveWallet.TransactionType.ETHSwap
+        ? getETHSwapTransactionBuyAndSellTokens({
+          nativeAsset: makeNetworkAsset(txNetwork),
+          tokensList: combinedTokensList,
+          tx: transaction
+        })
+        : {
+          buyToken: undefined,
+          sellToken: txToken,
+          buyAmount: new Amount(''),
+          sellAmount: new Amount(''),
+          buyAmountWei: new Amount('')
+        }
+    }, [
+      txNetwork,
+      combinedTokensList,
+      transaction,
+      txToken]
+    )
+
+    const [
+      normalizedTransferredValue,
+      transferredValueWei
+    ] = React.useMemo(() => {
+      const { normalized, wei } = getTransactionTransferredValue({
+        tx: transaction,
+        sellToken,
+        token: txToken,
+        txAccount: account,
+        txNetwork
+      })
+      return [normalized.format(6), wei]
+    }, [
+      transaction,
+      sellToken,
+      txToken,
+      account,
+      txNetwork
+    ])
+
+    const formattedSendCurrencyTotal =
+      getTransactionFormattedSendCurrencyTotal({
+        normalizedTransferredValue,
+        tx: transaction,
+        sellToken,
+        token: txToken,
+        txNetwork
+      })
+
+    const networkAssetPriceId =
+      networkAsset
+        ? getPriceIdForToken(networkAsset)
+        : ''
+    const txTokenPriceId =
+      txToken
+        ? getPriceIdForToken(txToken)
+        : ''
+    const sellTokenPriceId =
+      sellToken
+        ? getPriceIdForToken(sellToken)
+        : ''
+    const buyTokenPriceId =
+      buyToken
+        ? getPriceIdForToken(buyToken)
+        : ''
+    const priceIds =
+      [
+        networkAssetPriceId,
+        txTokenPriceId,
+        sellTokenPriceId,
+        buyTokenPriceId
+      ].filter((token) => token !== '')
```

Comment:
How about this to prevent defining a new function?

```suggestion
      ].filter(Boolean)
```

Reply:
Good idea, updated